src/long.ts

```diff
@@ -1021,9 +1023,37 @@ export class Long {
     if (options && options.relaxed) return this.toNumber();
     return { $numberLong: this.toString() };
   }
-  static fromExtendedJSON(doc: { $numberLong: string }, options?: EJSONOptions): number | Long {
-    const result = Long.fromString(doc.$numberLong);
-    return options && options.relaxed ? result.toNumber() : result;
+  static fromExtendedJSON(
+    doc: { $numberLong: string },
+    options?: EJSONOptions
+  ): number | Long | bigint {
+    const defaults = { useBigInt64: false, relaxed: true };
+    const ejsonOptions = { ...defaults, ...options };
```

Comment:
```suggestion
    const { useBigInt64 = false, relaxed = true } = { ...options };
```

Something to consider, removed the need for the temp defaults object